src/GraphQL.DataLoader/DataLoaderPair.cs

```diff
@@ -60,15 +73,22 @@ public void SetResult(T value)
         /// <param name="cancellationToken">Optional <seealso cref="CancellationToken"/> to pass to fetch delegate</param>
         public async Task<T> GetResultAsync(CancellationToken cancellationToken = default)
         {
-            if (!IsResultSet)
-                await Loader.DispatchAsync(cancellationToken).ConfigureAwait(false);
+            var loader = _loader;
+            if (loader != null)
+            {
+                // it does not matter if there are simultaneous calls to DispatchAsync as DataLoaderList
+                // protects against double calls to DispatchAsync
+                await loader.DispatchAsync(cancellationToken).ConfigureAwait(false);
+            }
             return Result;
         }
 
         async Task<object?> IDataLoaderResult.GetResultAsync(CancellationToken cancellationToken)
         {
-            if (!IsResultSet)
-                await Loader.DispatchAsync(cancellationToken).ConfigureAwait(false);
+            // same code as above; prevents an additional allocation
```

Comment:
> prevents an additional allocation

where?

Reply:
If GetResultAsync calls the other overload, I believe it will cause an allocation while re-wrapping the value in a new Task.